plugins/miscTags/miscTags.py

```diff
@@ -0,0 +1,154 @@
+                    new_scene['tag_ids'].append(tags_cache[t])
+                    update=True
+            if update:
+                log.info('Adding tags to scene: %s, tags: %s' % (scene['title'], tags,))
+                stash.update_scene(new_scene)
+        else:
+            log.debug('no update')
+    else:
+        log.debug('skipping scene')
+
+
+def processStashVRCompanionTags(scene,tags):
+    found=False
+    for f in scene['files']:
+        for k,v in VRCTags.items():
+            if k in f['basename'].lower():
+                tags.extend(v['VRCTags'])
+                found=True
+    if found:
+        tags.append('export_deovr')
+    return None
+
+def processVRTags(scene,tags):
+    found=False
+    for f in scene['files']:
+        for k,v in VRCTags.items():
+            if k in f['basename'].lower():
+                tags.extend(v['projTags'])
+                found=True
+    if found:
+        if 'vrTag' in stash.get_configuration()['ui']:
+            vr_tag=stash.get_configuration()['ui']['vrTag']
+            if vr_tag:
+                tags.append(vr_tag)
+            else:
+                tags.append('VR')
+        else:
+            tags.append('VR')
+    return None
+
+def processFlatStudio(scene,tags):
+    log.debug(scene)
+    if scene['studio']:
+        if scene['studio']['id'] in [x.strip() for x in settings['flatStudio'].split(',')]:
+            if 'export_deovr' not in tags:
+                tags.append('export_deovr')
+
+                tags.append('FLAT')
+    log.debug(tags)
+
+def processScenes():
+    log.info('Getting scene count')
+    if skip_tag not in tags_cache:
+        tags_cache[skip_tag]=stash.find_tag(skip_tag, create=True).get("id")
+    count=stash.find_scenes(f={"tags":{"depth":0,"excludes":[tags_cache[skip_tag]],"modifier":"INCLUDES_ALL","value":[]}},filter={"per_page": 1},get_count=True)[0]
+    log.info(str(count)+' scenes to process.')
+    i=0
+    for r in range(1,int(count/per_page)+2):
+        log.info('adding tags to scenes: %s - %s %0.1f%%' % ((r - 1) * per_page,r * per_page,(i/count)*100,))
+        scenes=stash.find_scenes(f={"tags":{"depth":0,"excludes":[tags_cache[skip_tag]],"modifier":"INCLUDES_ALL","value":[]}},filter={"page":r,"per_page": per_page})
+        for s in scenes:
+            processScene(s)
+            i=i+1
+            log.progress((i/count))
+            time.sleep(1)
+
+
+json_input = json.loads(sys.stdin.read())
+
+FRAGMENT_SERVER = json_input["server_connection"]
+stash = StashInterface(FRAGMENT_SERVER)
+config=stash.get_configuration()['plugins']
+if 'misc-tags' in config:
+    settings.update(config['misc-tags'])
+log.info('config: %s ' % (settings,))
+
+if 'mode' in json_input['args']:
+    PLUGIN_ARGS = json_input['args']["mode"]
+    if 'processScenes' in PLUGIN_ARGS:
+        processScenes()
+    elif 'test' in PLUGIN_ARGS:
+        log.debug(stash.get_configuration()['ui']['vrTag'])
```

Comment:
This will crash the user hasn't configured their VR tag, maybe print a nice message instead? Probably not necessary, just a nitpick 😅

Reply:
I'e updated the code